tests/transport_test.go

```diff
@@ -0,0 +1,118 @@
+package tests
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/pkg/errors"
+	"kubevirt.io/containerized-data-importer/pkg/common"
+	"kubevirt.io/containerized-data-importer/tests/framework"
+	"kubevirt.io/containerized-data-importer/tests/utils"
+
+	"k8s.io/api/core/v1"
+)
+
+var _ = Describe("Transport Tests", func() {
+
+	const (
+		annEndpoint = "cdi.kubevirt.io/storage.import.endpoint"
+		annSecret   = "cdi.kubevirt.io/storage.import.secretName"
+
+		fileHostName   = "cdi-file-host" // deployment and service name
+		fileHostNs     = "kube-system"   // deployment and service namespace
+		fileHostUname  = "admin"         // http and s3 username
+		fileHostPasswd = "password"      // http and s3 password
+
+		httpPortName = "http"
+		s3PortName   = "s3"
+
+		targetFile = "tinyCore.iso"
+		bucket     = "images"
+	)
+
+	var (
+		httpEp, s3Ep, ns string
+		fileHostService  *v1.Service // persistent file host service, routes to http and s3 ports
+	)
+
+	f, err := framework.NewFramework("", framework.Config{false})
+	if err != nil {
+		Fail(errors.Wrap(err, "Failed to create test framework").Error())
+	}
+
+	c, err := f.GetKubeClient()
+	if err != nil {
+		Fail(errors.Wrap(err, "Failed to create k8s client").Error())
+	}
+
+	cdiLabel := map[string]string{common.CDI_LABEL_KEY: common.CDI_LABEL_VALUE}
+
+	BeforeEach(func() {
+		ns = f.Namespace.Name
+
+		By(fmt.Sprintf("Waiting for all \"%s/%s\" deployment replicas to be Ready", fileHostNs, fileHostName))
```

Comment:
Do we need to reconstruct the service end point URL in each before each? The end points don't change so we really only need to construct them once, and put them somewhere every test can get at them.

Reply:
fixed